config/initializers/devise.rb

```diff
@@ -249,8 +249,8 @@
   # Add a new OmniAuth provider. Check the wiki for more information on setting
   # up on your models and hooks.
   # config.omniauth :github, 'APP_ID', 'APP_SECRET', scope: 'user,public_repo'
-  Devise.setup do |config|
-    config.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
+  Devise.setup do |con|
+    con.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
```

Comment:
You're fixing the symptom rather than the cause. Nesting `Devise.setup` blocks one inside the other makes no sense. Extract the content of the inner one, and put it in the outer one.